### PR TITLE
[BUILD] Allow use custom ARROW_HOME for Velox

### PR DESCRIPTION
**Pull request by Yohahaha**
If we use custom Arrow code path and build it before Velox, Velox could use pre-build libs.

### CODE REVIEW

**Comment by Yohahaha** [comment]
Hi @zhejiangxiaomai Would you help review this?

**Review by zhejiangxiaomai — APPROVED** [review]
LGTM

**Comment by Yohahaha** [comment]
Below case failed in CI, but this PR should not cause any UT failed.

```
- should be able to resolve a persistent view

java.lang.ArithmeticException: Decimal precision 20 exceeds max precision 2
  	at org.apache.spark.sql.errors.QueryExecutionErrors$.decimalPrecisionExceedsMaxPrecisionError(QueryExecutionErrors.scala:1013)
  	at org.apache.spark.sql.types.Decimal.set(Decimal.scala:123)
  	at org.apache.spark.sql.types.Decimal$.apply(Decimal.scala:578)
  	at org.apache.spark.sql.types.Decimal.apply(Decimal.scala)
  	at io.glutenproject.vectorized.ArrowWritableColumnVector$DecimalAccessor.getDecimal(ArrowWritableColumnVector.java:1010)
  	at io.glutenproject.vectorized.ArrowWritableColumnVector.getDecimal(ArrowWritableColumnVector.java:728)
  	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
```

**Comment by zhejiangxiaomai** [comment]
Thanks, I rerun the failed CI. will merge it when CI passed.

File: ep/build-velox/src/build_velox.sh [hunk 1]
```diff
@@ -76,9 +76,9 @@ function compile {
   # Let Velox use pre-build arrow,parquet,thrift.
   if [ "$ARROW_HOME" == "" ]; then
     ARROW_HOME=$CURRENT_DIR/../../build-arrow/build
-    if [ -d "$ARROW_HOME" ]; then
-      COMPILE_OPTION="$COMPILE_OPTION -DArrow_HOME=${ARROW_HOME}"
-    fi
+  fi
+  if [ -d "$ARROW_HOME" ]; then
+    COMPILE_OPTION="$COMPILE_OPTION -DArrow_HOME=${ARROW_HOME}"
```
Comment (jinchengchenghh) [inline ep/build-velox/src/build_velox.sh:81]: -DArrow_HOME -> -DARROW_HOME

**Pull request merged**